src/DotNet/Pdb/PdbState.cs

```diff
@@ -133,6 +134,8 @@ public sealed class PdbState {
 			// Expensive part, can read source code etc
 			doc.Initialize(symDoc);
 			docDict.Add(doc, doc);
+			if (symDoc.MDToken.HasValue)
+				tokenToDocument.Add(symDoc.MDToken.Value, doc);
```

Comment:
Need to update tokenToDocument in these methods too:

- Add_NoLock(PdbDocument doc)
- Remove(PdbDocument doc)
- RemoveAllDocuments(bool returnDocs)